### PR TITLE
Create Total Activities Component for End of Unit pages

### PR DESCRIPTION
**Pull request by ianlaidlaw**
## Relevant Rally Links

- https://rally1.rallydev.com/#/289692574792d/dashboard?detail=%2Fuserstory%2F617359155011&fdp=true

## Description

This pr creates a new component that will display how many of required or optional topics are in a unit. This component is not in use yet, but will be as part of the end of unit redesigns. There exists a subEntity on the "end of sequence" entity that contains these numbers and will spit them out.

This component will only render if any optional topics exist, per the design.

## Video/Screenshots

image

image

image

## Related PRs

- lms:
- sequences:

### CODE REVIEW

File: d2l-sequence-navigator/d2l-missed-activity.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ class D2LMissedActivity extends ASVFocusWithinMixin(PolymerASVLaunchMixin(Comple
 		return {
 			numberOfMissedActivity:{
 				type: String,
-				computed: 'getnumberOfMissedActivity(subEntities)'
+				computed: 'getNumberOfMissedActivity(subEntities)'
```
Comment (ianlaidlaw) [inline d2l-sequence-navigator/d2l-missed-activity.js:80]: unrelated refactor

File: helpers/hypermedia-helper.js [hunk 2]
```diff
@@ -116,12 +116,12 @@ class HypermediaHelper {
 		}
 	}
 
-	static getTopicAndModuleSubEntities(entity) {
+	static getTopicAndModuleSubEntities(entity, rel = '[\\s\\S]*') {
```
Comment (ianlaidlaw) [inline helpers/hypermedia-helper.js:119]: we now have subEntities with different rels, so I kept the "all subentities" regexp as a default, but we can optionally pass one in

**Review by coffeexcode — APPROVED** [review]
Looks good!

File: lang/en.json [hunk 3]
```diff
@@ -40,6 +40,9 @@
     "openNew": "Open in a new window",
     "openOnedriveFile": "Open OneDrive file in a new browser tab.",
     "openUnsecureContent": "Your browser thinks this content is published in a less secure location. Don't worry, just open the content in a new window.",
+	"optional": "optional",
+	"optionalSummary": "Optional Activities are not required for the completion of this Unit",
```
Comment (Kiefer-Co-D2L) [inline lang/en.json:44]: Should Activities be capitalized here (and in `locales.json`)?  The demo pic has it as lowercase.

Comment (ianlaidlaw) [reply lang/en.json:44]: yep
